lua/entities/gmod_wire_expression2/core/custom/sv_rune2.lua

```diff
@@ -20,69 +21,54 @@ local function buildBody(args) -- Why does the wireteam do this
 end
 
 -- We use this for try and catch
-local function runE2InstanceSafe(compiler,func,body) -- Varargs to pass to the e2 function
-    -- Will always return pcallerror,errstr first even if it didn't error.
-    local args = {pcall(func,compiler,body)}
+local function runE2InstanceSafe(compiler,func,body,...)
+    -- Always return pcallerror,errstr first even if it didn't error.
+    local args = {pcall(func,compiler,body,...)}
     local success = table_remove(args,1)
     if success then
-        return true,"none",args
-    else
-        -- Don't return args, that would be a waste
-        local errmsg = table_remove(args,1)
-        if errmsg == "exit" then return true,"none",{} end -- nice exit()
-        if errmsg == "perf" then errmsg = "tick quota exceeded" end
-        return false,errmsg
+        return true,nil,args
     end
+    -- If unsuccessful, don't return args, that would be a waste
+    local errmsg = table_remove(args,1)
+    if errmsg == "exit" then return true,nil,{} end -- nice exit()
+    if errmsg == "perf" then errmsg = "tick quota exceeded" end
+    return false,errmsg
 end
 
-local function getE2FuncFromStr(compiler,funcname)
-    local funcs = compiler.funcs
-    local e2func = funcs[funcname]
-    if not e2func then -- We will look for any udfs that have the name before the parenthesis.
-        local fncnameproper = funcname:match("(%w*)%(") or funcname
-        for K,V in pairs(funcs) do
-            local proper = K:match("(%w*)%(")
-            if proper and string.find(proper,fncnameproper) then return V end
-        end
-    else
-        return e2func
-    end
-end
+-- TODO: Set E2 costs ( __e2setcost(N) ).
```

Comment:
You forgot set E2 cost for these functions. We can determine the costs and fix this later. This is just a reminder.